adam/experiment/observer.py

```diff
@@ -786,6 +787,31 @@ def from_params(
             file_name=params.optional_string("file_name"),
         )
 
+    def _convert_to_output_format(
+        self,
+        idx: int,
+        semantic_node: SemanticNode,
+        linguistic_description: LinguisticDescription,
+        predicted_scene_description: TopLevelLanguageLearnerDescribeReturn,
+    ) -> Mapping[str, Any]:
+        if isinstance(semantic_node, (ActionSemanticNode, RelationSemanticNode)):
+            return_type = "complex"
+        else:
+            return_type = "object"
```

Comment:
Could consider using a syntax like this:
`return_type = "complex" if isinstance(semantic_node, (ActionSemanticNode, RelationSemanticNode)) else "object"`